### PR TITLE
Golang

### PR DESCRIPTION
**Pull request by psykokwak4**
Add CodecService/Provider and tests

### CODE REVIEW

**Comment by abhikeshav** [comment]
Thanks, Xiaoqin! Just wondering why the commits seems upside down in order?

**Comment by psykokwak4** [comment]
@abhikeshav Merge conflicts, almost all commits in my repository's golang branch differ from commits in DevNet/golang(probably because of force update?), check out the squashed [commit](https://github.com/CiscoDevNet/ydk-gen/pull/484/commits/8e44e3c67ac02850a70318dfaa521853840b7da6).

**Comment by abhikeshav** [comment]
Hmm.. interesting. I did see the full commit history. It was just upside down. Now I just see one commit

File: sdk/go/core/ydk/registry/registry.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *----------------------------------------------
+ */
+package registry
```
Comment (abhikeshav) [inline sdk/go/core/ydk/registry/registry.go]: Can we keep the naming consistent across ydk-cpp and go? We call this `entity-lookup` in C++

**Pull request merged**